openreview/arr/helpers.py

```diff
@@ -1467,106 +1466,135 @@ def flag_submission(
     short_name = domain.get_content_value('subtitle')
     forum = client.get_note(id=edit.note.forum, details='replies')
 
-    ethics_flag_field = list(flagging_info['ethics_flag_field'].keys())[0]
-    violation_fields = list(flagging_info['violation_fields'].keys())
-    violation_default = flagging_info['violation_fields']
-    ethics_flag_default = flagging_info['ethics_flag_field'][ethics_flag_field]
-    reply_name = flagging_info['reply_name']
-
-    ethics_flag_edits = client.get_note_edits(note_id=edit.note.forum, invitation=f"{venue_id}/-/Ethics_Review_Flag")
-    dsv_flag_edits = client.get_note_edits(note_id=edit.note.forum, invitation=f"{venue_id}/-/Desk_Reject_Verification_Flag")
+    ethics_flag_default = 'No'
+    ethics_flag_fields = {
+        'Review': 'needs_ethics_review',
+        'Checklist': 'need_ethics_review'
+    }
+    violation_fields = {
+        'Checklist': {
+            'appropriateness': 'Yes',
+            'formatting': 'Yes',
+            'length': 'Yes',
+            'anonymity': 'Yes',
+            'responsible_checklist': 'Yes',
+            'limitations': 'Yes'
+        },
+        'Official_Review': {
+            'Knowledge_of_or_educated_guess_at_author_identity': 'No'
+        },
+        'Meta_Review': {
+            'author_identity_guess': [4, 3, 2, 1]
+        }
+    }
 
-    dsv_flagged = forum.content.get('flagged_for_desk_reject_verification', {}).get('value')
-    ethics_flagged = forum.content.get('flagged_for_ethics_review', {}).get('value')
-    has_ethic_flag_history = len(ethics_flag_edits) > 0
-    has_dsv_flag_history = len(dsv_flag_edits) > 0
+    dsv_flagged = forum.content.get('flagged_for_desk_reject_verification', {}).get('value', False)
+    ethics_flagged = forum.content.get('flagged_for_ethics_review', {}).get('value', False)
 
     def post_flag(invitation_name, value=False):
-       return client.post_note_edit(
+        print(f"posting {value} to {invitation_name}")
+        return client.post_note_edit(
             invitation=f'{venue_id}/-/{invitation_name}_Flag',
             note=openreview.api.Note(
                 id=edit.note.forum,
                 content={f'flagged_for_{invitation_name.lower()}': {'value': value}}
             ),
             signatures=[venue_id]
         )
+
+    def check_field_violated(note, field, default_value):
```

Comment:
@haroldrubio I am a little bit confused about what this function does and why we have to check this?

Reply:
I think this function is the one that checks in the content of the note if there is an indicator that the submission needs ethics reviews:

for official reviews, it checks the field `needs_ethics_review` violates the default value that is 'No'

Reply:
Yes that's correct it's just a small helper function for that

Reply:
I see, thanks! I was also confused because of the `violation_fields` but now I see those are to check for desk reject verification